modules/host/host.go

```diff
@@ -148,6 +158,38 @@ func (h *Host) Capacity() int64 {
 	return h.spaceRemaining
 }
 
+// Close shuts down the host, preparing it for garbage collection.
+func (h *Host) Close() error {
+	// The order in which things are closed has been explicitly chosen to
+	// minimize turbulence in the event of an error.
+
+	// Save the latest host state.
+	err := func() error {
+		h.mu.Lock()
+		defer h.mu.Unlock()
+		return h.save()
+	}()
```

Comment:
I've always done the latter, e.g. in host/negotiate.go [here](https://github.com/NebulousLabs/Sia/blob/46ce037b89bb4e6ea1adbee1937e41969ac69715/modules/host/negotiate.go#L195), [here](https://github.com/NebulousLabs/Sia/blob/46ce037b89bb4e6ea1adbee1937e41969ac69715/modules/host/negotiate.go#L298), and [here](https://github.com/NebulousLabs/Sia/blob/46ce037b89bb4e6ea1adbee1937e41969ac69715/modules/host/negotiate.go#L346). I thought we figured out a while ago that a good pattern is "gather data under lock, branch out of lock." Sometimes this isn't feasible, but 90% of the time it is. As far as I'm aware, the former pattern in your comment is relatively uncommon in our code now.
I think a simple convention of "avoid branching under locks" is better than rewriting half of the project's locking code in an unintuitive new style.